File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        max_val = (2 ** 31) - 1
        sign = -1 if x < 0 else 1
        x = abs(x)
        rev = 0
        while x != 0:
            digit = x % 10
            x /= 10
            rev = rev * 10 + digit
            if rev > (max_val-digit/10):
                return 0

        return rev*sign

        

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        // Step 1: Get the sign
        int sign = x < 0 ? -1 : 1;

        // Step 2: Convert absolute value to string
        String str = new StringBuilder(String.valueOf(Math.abs((long)x))).reverse().toString();

        // Step 3: Convert back to integer and handle overflow
        try {
            int reversed = Integer.parseInt(str);
            return reversed * sign;
        } catch (NumberFormatException e) {
            // If it overflows
            return 0;
        }
    }
}
